app/assets/javascripts/map.js

```diff
@@ -300,10 +300,12 @@ window.getColor = function getColor(d) {
 
 var styleFce = function styleFce(f) {
   var style = defaultStyle;
-  style.fillColor = f.tags.legend_color || '#fff';
-  if (style.fillColor == '#fff') {
-    console.log(f.tags);
-  }
+
+  all_tags = [];
+  f.tags.snippets.map(function(elm) { all_tags = all_tags.concat(elm.tags); })
+
+  style.fillColor = CategoryLegend.get_color_by_tags(all_tags);
```

Comment:
This calculation is really slow. Has to be improved:
* do not use temporary variable (call directly category_legend.get_color_by_tags(properties)
* cache result (maybe directly in the `f.tags.legend_color[tags_filter]` and `CategorLegend.selected_values`) 

btw:
* now tags_filter is stored in two places: `params['q[tags_filter]']` and CategoryLegend.
* use `var properties = f.properties || f.tags` in GeoJSON is normally used `properties` but getjson-vs uses `tags`;